src/material/core/common-behaviors/common-module.ts

```diff
@@ -75,7 +93,10 @@ export class MatCommonModule {
   }
 
   private _checkDoctypeIsDefined(): void {
-    if (this._document && !this._document.doctype) {
+    const isEnabled = this._checksAreEnabled() &&
+      (typeof this._sanityChecks === 'boolean' ? this._sanityChecks : this._sanityChecks.doctype);
```

Comment:
Rather than going to runtime typeof, [what do you think of doing this](https://www.typescriptlang.org/play/?noEmitHelpers=true#code/PQKhCgAIUgBAbAlgYwKYDsDOqowMID2ADgJ4BOiA5gBYAukA4gQZfKpADId6QCC88SACUqdTMNTYyAN1QATAHS5ckAKrZIBAGaRa1ROMwEArmTSRkBOewORKBWWXTzIAIxKQAhukgBZAJIAKgC0mLQkbJBIaFjsep70yN5uONCQWibocpCIPnrsHP54AKIAcgDKxemIkQmQdLREmABcwMDelMbwnmQKiATA0RjYuMDg4G2QAAoU0gnsjpj9PpZYtN70tAQWiGbGALay6JuStMCuxjXZBpjGkgA0E8A5CqgKXtIEiNmWZHHb1iIGGyBDy1HYsA6XR6wH28wonkERAocLIHgwtDRwSIX2OT0geGSrnYf32DhcoPM+T88MQiMgyMQqPRxyxONy9HQ23goMoqDI+L+wVQAA8cWRaOJEYJsKs5D0WZiSNjcZKJmA4HICMhMNjZvNoGNwFpjOhkLRljlMAARVDSXxWVAACgAlM03Mw2MkAN5QSB-WimPJkO4AbnAAF9xqAIGkpgRMEtXJFMN5EOELODkABrcTxRJE9gYTzJ+Tvfw6bCbAgqTF3YBaRHYe5eASZ1A58QAdxqgmJkGLpbkwDkBhLbEUo3AovFmxIQMg5TT4TwWdzkAAvJBfZBIFrzfPUO7XJ7UN5w7v8vsjx6CF70BfIItlsfT+e-dRPPtr2RX3ezw+kbhuqsYwA6chdHEn4FkQgZ-FK6AkHouSULo0GQJg1AmPA2T9jynjWNk3jDgQZD+qa6SkdA0owLw6CdN0ZG+LSiIqJY+w4s4xyYO8gT6OIuTIPAxjWOIABC3yIC2qC0MgSjQCofG2GSEEplhXS4TYHGkbQLhbAOnjINQujEMEbCyII7GcRikAqZBkBOm8lAKC2zG0IEJaYOBkEuvJIBjDOOkWN0iY0rQhDfqC3mRDukAxpAADq4L5GRXaoAA5LIe6glB7CsAQrj0qm6Dph4RkduujkKM5Xhoag15WlEBAEfILYpTYUrZQeQK+YafqMnMumQAA+p+No5QwPKFfAq4VeIW6NvA2DAbu8VCKgWj8hgVLbNS+XTZAAAG+4HBih2aK4ABWHa0O8-n9fqQ3DSd17HJuuiHtoXWnW9G5-ZA6UFdd5rpZAABkYPfa99AAPxQzZ7roF08ArXFGrrZtfxmv8dV2FN9LpT2WQEF2oNAzdd1jLuA0GsNRNal273hECX30yTm7-YDV03aDEOQGzjNwwLkCI8jqPxYQ6BaFQpguMVpXtp2lMPYgg3sMN8srmuLSLsuJCzZ2kAAD63veqOrGEIbmqRTqa-r2vut4JAutu4y7qtzylAQQ35kdGt6wbubnbYzN6dsh1O4dLbEkkxgaLwBCBH6HtoWQJMIQOZBp2RiA6GlkBx7lR1LiVWtzedod7rsN3wCQ8nu8hPH+6X9tze9duB5gqO7rnDkAISNwoo2eONziTQViKdy7sXu2hBhD+VObWtqof+DaG25PIrqPrPg-D6PqDj9NndMyGqA75AUa7lGfrxUlMngmRTsYXrivrj0RboOOZZ9dTj3q4vXMvA-jFC-kOV0f57yu1ngGIMVpbT2kdK6cGkMB78SHgYQIpxQHSG3n6G+q0NT3xSrjSw1hGohnQCVeiOQ8inB4r-Bk-8RqYOwegXB09k4WFBGEfmuQGZM3QXTfh7MR5eEQhfWBTg+HE0ZnzJ0Ash7DWzD0OEw1hrGxNooy6I99ibxdI+AhTDVa00AcvbqqA162mls4OQECnxfGyDPbhawrSgO-tkLce9AGYGAagdx4CXYQy4buJ0e8O7aw5l4s+miHLhIDpEsRJdSpTwUPuUOBi3bu17k6AwASJwoLnk3F6Nk+ZoPns9bUP1brpMPJw2eu4Lb-gUF2Ho6AnQhPdulPApgsb0BKW9LUkhIBcnoJ+LKnguqh14vxWyngyqeELgDSAABqTpu50pGAanRBiPQwr8jpJZAg2kuKShGd7EyKRxnsDETOG6ZZ0qdMybPK+l9IxAA).